From 79bee1dfb9eafe2a0e031743f70db0c0a72cacb4 Mon Sep 17 00:00:00 2001
From: Sinthu Raja <sinthu.raja@mistralsolutions.com>
Date: Wed, 14 Jul 2021 18:22:20 +0530
Subject: [tiL5.10-CON PATCH 2/3] media: ti: j721e-csi2rx: Add RAW8 and RAW10
 pixel format support

Extend the list of supported image formats to include all the RGB layouts
of RAW8 and RAW10.

Signed-off-by: Sinthu Raja <sinthu.raja@ti.com>

- Change bpp for RAW10 formats to 16.
- Use small letters for hex in RAW8 and RAW10 data format definitions.
- Set size shift for all the formats.
Signed-off-by: Pratyush Yadav <p.yadav@ti.com>
---
 .../platform/ti/j721e-csi2rx/j721e-csi2rx.c   | 58 +++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c b/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
index 9e92be7fbaf1..2129545c6119 100644
--- a/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
+++ b/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
@@ -44,6 +44,8 @@
 #define CSI_DF_YUV422			0x1e
 #define CSI_DF_RGB444			0x20
 #define CSI_DF_RGB888			0x24
+#define CSI_DF_RAW8			0x2a
+#define CSI_DF_RAW10			0x2b
 
 #define PSIL_WORD_SIZE_BYTES		16
 #define TI_CSI2RX_NUM_CTX		16
@@ -163,6 +165,62 @@ static const struct ti_csi2rx_fmt formats[] = {
 		.csi_df			= CSI_DF_YUV422,
 		.bpp			= 16,
 		.size			= SHIM_DMACNTX_SIZE_8,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SBGGR8,
+		.code			= MEDIA_BUS_FMT_SBGGR8_1X8,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW8,
+		.bpp			= 8,
+		.size			= SHIM_DMACNTX_SIZE_8,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SGBRG8,
+		.code			= MEDIA_BUS_FMT_SGBRG8_1X8,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW8,
+		.bpp			= 8,
+		.size			= SHIM_DMACNTX_SIZE_8,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SGRBG8,
+		.code			= MEDIA_BUS_FMT_SGRBG8_1X8,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW8,
+		.bpp			= 8,
+		.size			= SHIM_DMACNTX_SIZE_8,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SRGGB8,
+		.code			= MEDIA_BUS_FMT_SRGGB8_1X8,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW8,
+		.bpp			= 8,
+		.size			= SHIM_DMACNTX_SIZE_8,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SBGGR10,
+		.code			= MEDIA_BUS_FMT_SBGGR10_1X10,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW10,
+		.bpp			= 16,
+		.size			= SHIM_DMACNTX_SIZE_16,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SGBRG10,
+		.code			= MEDIA_BUS_FMT_SGBRG10_1X10,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW10,
+		.bpp			= 16,
+		.size			= SHIM_DMACNTX_SIZE_16,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SGRBG10,
+		.code			= MEDIA_BUS_FMT_SGRBG10_1X10,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW10,
+		.bpp			= 16,
+		.size			= SHIM_DMACNTX_SIZE_16,
+	}, {
+		.fourcc			= V4L2_PIX_FMT_SRGGB10,
+		.code			= MEDIA_BUS_FMT_SRGGB10_1X10,
+		.colorspace		= V4L2_COLORSPACE_SRGB,
+		.csi_df			= CSI_DF_RAW10,
+		.bpp			= 16,
+		.size			= SHIM_DMACNTX_SIZE_16,
 	},
 
 	/* More formats can be supported but they are not listed for now. */
-- 
2.34.1

